checker/src/types/store.rs

```diff
@@ -120,6 +120,8 @@ impl Default for TypeStore {
 				to: TypeId::T_TYPE,
 				parameters: Some(vec![TypeId::T_TYPE]),
 			},
+
+			Type::Class { name: "ImportMeta".to_owned(), parameters: None },
```

Comment:
This is what happen when I use interface:
```
cargo run -p ezno-checker -F ezno-parser --example cache ./checker/definitions/overrides.d.ts ./checker/definitions/internal.ts.d.bin
   Compiling ezno-checker v0.0.15 (/home/apika/Downloads/ezno/checker)
    Finished dev [unoptimized + debuginfo] target(s) in 3.96s
     Running `target/debug/examples/cache ./checker/definitions/overrides.d.ts ./checker/definitions/internal.ts.d.bin`
thread 'main' panicked at /home/apika/Downloads/ezno/checker/src/synthesis/classes.rs:398:9:
internal error: entered unreachable code: expected class type Interface { name: "ImportMeta", nominal: false, parameters: None }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Related to this:
```typescript
declare class ImportMeta {
  static env: ImportEnv;
  static url: string;
  static resolve(url: string): string;
}
```

But if I try to use an `interface` instead of `class`, it tells that `Interface` is not implemented for definition file:
```
thread 'main' panicked at /home/apika/Downloads/ezno/checker/src/lib.rs:643:5:
found error in definition file [
    Position {
        reason: "Expected Keyword(Let), Keyword(Const), Keyword(Var), Keyword(Function), Keyword(Class) or Keyword(Type) found Keyword(Interface)",
        position: 140..149#1,
        kind: Error,
    },
]
```

So, for now I keep it with `Class`

Reply:
I've found how how do it work with interface, but now it doesn't get the properties right:
```
error: 
  ┌─ a.ts:1:13
  │
1 │ const env = import.meta.env;
  │             ^^^^^^^^^^^^^^^ No property 'env' on ImportMeta

note: 
  ┌─ a.ts:3:1
  │
3 │ debug_type(env);
  │ ^^^^^^^^^^^^^^^ Types: error
```

Reply:
Hmm will look into that

Reply:
Awesome, my reasoning for using `interface` instead of class is `Array` exists in JS as a type **and** as `new Array` and static methods like `Array.from` etc. And because Array is a class that exists in runtime but isn't a user written class I used `declare class`. 

With `ImportMeta` you can't do `new ImportMeta` or `ImportMeta.something`. (thinking about it now maybe `Math` shouldn't be a class). And as all `interface`s are declared you don't need the keyword. It's a little bit complicated but am trying to keep to TS conventions 😅.

Yes I think the problem was that for the checker to work all the internal types need to be the same. So you had it in the `.d.ts` file as a `interface` and the stored internal types where expecting a `Class` and so it wasn't connect the dots https://github.com/kaleidawave/ezno/blob/bfd2133287e68a8b32016b208dbcac18629c4eec/checker/src/context/mod.rs#L723-L724

But it all works now and the test is passing 🎉 https://github.com/Brayan-724/ezno/blob/main/checker/specification/specification.md#importmeta